library/src/scala/NamedTuple.scala

```diff
@@ -0,0 +1,184 @@
+package scala
+import annotation.experimental
+import compiletime.ops.boolean.*
+
+@experimental
+object NamedTuple:
+
+  opaque type AnyNamedTuple = Any
+  opaque type NamedTuple[N <: Tuple, +V <: Tuple] >: V <: AnyNamedTuple = V
+
+  def apply[N <: Tuple, V <: Tuple](x: V) = x
```

Comment:
This really must have an explicit result type. As defined it returns a `V`, not a `NamedTyple[N, V]`!
```suggestion
  def apply[N <: Tuple, V <: Tuple](x: V): NamedTuple[N, V] = x
```
In general, *all* public APIs should have explicit result types.

The fact that this was not caught by tests is worrying.

Reply:
The result type is `V` not `NamedTuple[N, V]`. Agree it should be added.